minorminer/utils/parallel_embeddings.py

```diff
@@ -0,0 +1,630 @@
+                    q: idx if n in node_set else float("nan")
+                    for idx, emb in enumerate(_embeddings)
+                    for n, c in emb.items()
+                    for q in c
+                }
+            )
+        else:
+            node_color_dict.update(
+                {
+                    q: idx
+                    for idx, emb in enumerate(_embeddings)
+                    for n, q in emb.items()
+                    if n in node_set
+                }
+            )
+    # Create edge color mapping
+    edge_color_dict = {}
+    if S is not None:
+        edge_color_dict = {
+            (tu, tv): idx
+            for idx, emb in enumerate(_embeddings)
+            for u, v in S.edges()
+            if u in emb and v in emb
+            for tu in (emb[u] if one_to_iterable else [emb[u]])
+            for tv in (emb[v] if one_to_iterable else [emb[v]])
+            if G.has_edge(tu, tv)
+        }
+        if one_to_iterable:
+            # Feature enhancement? We could consider formatting these lines
+            # differently to distinguish chain couplers from logical couplers
+            for idx, emb in enumerate(_embeddings):
+                for chain in emb.values():
+                    Gchain = G.subgraph(chain)
+                    edge_color_dict.update({e: idx for e in Gchain.edges()})
+
+    else:
+        edge_color_dict = {
+            (v1, v2): node_color_dict[v1]
+            for v1, v2 in G.edges()
+            if node_color_dict[v1] == node_color_dict[v2]
+        }
+
+    # Default drawing arguments
+    draw_kwargs = {
+        "G": G,
+        "node_color": [node_color_dict[q] for q in G.nodes()],
+        "edge_color": "lightgrey",
+        "node_shape": "o",
+        "ax": ax,
+        "with_labels": False,
+        "width": 1,
+        "cmap": cmap,
+        "edge_cmap": cmap,
+        "node_size": 300 / np.sqrt(G.number_of_nodes()),
+    }
+    draw_kwargs.update(kwargs)
+
+    topology = G.graph.get("family")
+    # Draw the combined graph with color mappings
+    if topology == "chimera":
+        pos = dnx.chimera_layout(G)
+        dnx.draw_chimera(**draw_kwargs)
+    elif topology == "pegasus":
+        pos = dnx.pegasus_layout(G)
+        dnx.draw_pegasus(**draw_kwargs)
+    elif topology == "zephyr":
+        pos = dnx.zephyr_layout(G)
+        dnx.draw_zephyr(**draw_kwargs)
+    else:
+        pos = nx.spring_layout(G)
+        nx.draw_networkx(**draw_kwargs)
+    if len(edge_color_dict) > 0:
+        # Recolor specific edges on top of the original graph
+        nx.draw_networkx_edges(
+            G,
+            pos=pos,
+            edgelist=list(edge_color_dict.keys()),
+            edge_color=list(edge_color_dict.values()),
+            width=1,
+            edge_cmap=cmap,
+            ax=ax,
+        )
+    return node_color_dict, edge_color_dict
+
+
+def shuffle_graph(
+    G: nx.Graph, seed: Union[int, np.random.RandomState, np.random.Generator] = None
```

Comment:
Not the cleanest, but there should be an `Optional` type-hint added here.
```suggestion
    G: nx.Graph, seed: Optional[Union[int, np.random.RandomState, np.random.Generator]] = None
```